### PR TITLE
Java Client generator

### LINKED ISSUE #37: Code generation for Java client API

### PR DESCRIPTION
**Pull request by 2m**
Adds java client generator following the current scaladsl and javadsl conventions. This spilts up current Java Server templates into two: server and common, which is used by the new client template as well.

Fixes #37

### CODE REVIEW

**Comment by 2m** [comment]
I am going to reopen this PR as the repo rename seems to have broken it (new commnits pushed to the branch do not show up).

File: codegen/src/main/twirl/templates/JavaClient/Client.scala.txt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+  private @{service.name}Client(Channel channel, CallOptions options, Materializer mat, ExecutionContext ex) {
+    this.channel = channel;
+    this.options = options;
+    this.mat = mat;
+    this.ex = ex;
+  }
+
+  public static @{service.name}Client create(Channel channel, CallOptions options, Materializer mat, ExecutionContext ex) {
+    return new @{service.name}Client(channel, options, mat, ex);
```
[the remark this answers is not in the document]
Comment (2m) [reply codegen/src/main/twirl/templates/JavaClient/Client.scala.txt]: I kept the same logic as in the scala side, which has context as a second implicit parameter next to the materializer.

File: runtime/src/main/scala/akka/grpc/internal/AkkaGrpcGraphStage.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package akka.grpc.internal
+
+import akka.annotation.InternalApi
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import io.grpc.stub.StreamObserver
+
+@InternalApi
+private class AkkaGrpcGraphStage[I, O](name: String, operator: StreamObserver[O] => StreamObserver[I]) extends GraphStage[FlowShape[I, O]] {
```
[the remark this answers is not in the document]
Comment (renatocaval) [reply runtime/src/main/scala/akka/grpc/internal/AkkaGrpcGraphStage.scala]: Somehow I didn't see that message before. Sounds good to me.

File: codegen/src/main/twirl/templates/JavaClient/Client.scala.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (renatocaval) [reply codegen/src/main/twirl/templates/JavaClient/Client.scala.txt]: We still have that ticket about removing the `Channel` dependency. I re-assigned it to myself and was planning to start working on this tomorrow. Anyways, I guess we can change that later one.

**Pull request closed without merging**